### PR TITLE
GEOS-11279 allow one field multiple tabs

### PR DESCRIPTION
**Pull request by NielsCharlier**
[GEOS-11279](https://osgeo-org.atlassian.net/browse/GEOS-11279) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+
+public class CommaSeperatedDeserializer extends JsonDeserializer<List<String>> {
+
+    @Override
+    public List<String> deserialize(JsonParser p, DeserializationContext ctxt)
+            throws IOException, JacksonException {
+        return Lists.newArrayList(p.getText().split(","));
```
Comment (mprins) [inline src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java:20]: You may want to `trim()` the split result to remove whitespace

Comment (aaime) [reply src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java:20]: or split over ``"\\s*,\\s*"``

Comment (NielsCharlier) [reply src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java]: adjusted

File: doc/en/user/source/extensions/metadata/uiconfiguration.rst [hunk 2]
```diff
@@ -119,7 +119,8 @@ For example:
 tab
 ^^^
 Optionally, attributes may be displayed on separate tabs. All tabs must be listed under `tabs` in the main configuration. Then this
-property is used to assign each attribute to a tab, so that the custom metadata panel is divided in tabs:
+property is used to assign each attribute to one or more tab (seperated by comma),
```
Comment (mprins) [inline doc/en/user/source/extensions/metadata/uiconfiguration.rst:122]:
```suggestion
property is used to assign each attribute to one or more tab (separated by comma), 
```

**Review by mprins — COMMENTED** [review]
pls fix spelling of separated.
apart from that OK

File: src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+
+public class CommaSeperatedDeserializer extends JsonDeserializer<List<String>> {
```
[the remark this answers is not in the document]
Comment (NielsCharlier) [reply src/extension/metadata/src/main/java/org/geoserver/metadata/data/dto/impl/CommaSeperatedDeserializer.java]: Thanks!

**Pull request merged**